initialization/initialization.go

```diff
@@ -144,42 +204,16 @@ func (init *Initializer) isInitializing() bool {
 	return init.initializing
 }
 
-func (init *Initializer) Stop() error {
-	if !init.isInitializing() {
-		return ErrNotInitializing
-	}
-
-	close(init.stopChan)
-	if res := gpu.Stop(); res != gpu.StopResultOk {
-		return fmt.Errorf("gpu stop error: %s", res)
-	}
-
-	select {
-	case <-init.doneChan:
-	case <-time.After(5 * time.Second):
-		return errors.New("stop timeout")
-	}
-
-	return nil
-}
-
-func (init *Initializer) SessionNumLabelsWrittenChan() <-chan uint64 {
-	init.mtx.RLock()
-	defer init.mtx.RUnlock()
-	return init.numLabelsWrittenChan
-}
-
 func (init *Initializer) SessionNumLabelsWritten() uint64 {
 	return init.numLabelsWritten.Load()
 }
 
 func (init *Initializer) Reset() error {
-	if init.isInitializing() {
+	switch init.Status() {
+	case StatusInitializing:
 		return ErrCannotResetWhileInitializing
-	}
-
-	if err := init.VerifyStarted(); err != nil {
-		return err
+	case StatusError:
+		return fmt.Errorf("cannot determine status of initialization")
```

Comment:
`Status()` is meant to be called concurrently and I don't see an issue with the method if it is called by itself.

`Reset()` as you say can be a problem. I just remembered the `TryLock()` method of a `sync.Mutex`. I think by locking both `Initialize()` and `Reset()` behind the same mutex and using `TryLock()` over `Lock()` to be able to return early if the mutex is in use by the other function, maybe this even lets us get rid of the `initializing` field. What's your opinion?

Reply:
I think there are too many things to describe the same property (there are status and initializing), they can go out of sync. A quick fix is to lock `init.mtx` before checking the status to avoid racing with `init.Initialize()` and unlock it upon quitting.

Otherwise, the whole `Initializer` could be documented as not thread-safe and the user should use a Mutex if the user wants to use it concurrently. 

I'd go with the second approach but it requires more refactoring on the go-spacemesh side.

Reply:
The problem with locking `Status()` and `Initialize()` behind the same mutex is, that `Status()` must be able to be called while `Initialize()` is running.

I know this shouldn't be necessary because the caller of `Status()` should know if `Initialize()` is currently running or not, but changing that behaviour is a bigger effort that is out of scope of this PR.

I will slightly improve the current condition by using `TryLock()` instead of `Lock()` to catch if `Status()` should return `StatusInitializing`.

Reply:
`Reset()` still has the same issue. Nothing prevents Initialization to start after `init.Status()` returns.

Reply:
Added `TryLock()` to `Reset()` and implemented a test checking the functionality.